intera_interface/src/intera_interface/clicksmart_plate.py

```diff
@@ -247,6 +247,29 @@ def get_endpoint_info(self, endpoint_id=None):
         endpoint_id = self.endpoint_map.keys()[0] if endpoint_id is None else endpoint_id
         return (endpoint_id, self.endpoint_map[endpoint_id])
 
+    def send_configuration(self, config, timeout=5.0):
+        """
+        Send a new JSON EndEffector configuration to the device.
+
+        @type config: dict  # ( | json?)
+        @param config: new json config to save on device
+        @type timeout: float
+        @param timeout: timeout in seconds
+        """
+        rospy.loginfo("Reconfiguring ClickSmart...")
+
+        cmd = IOCommand('reconfigure', {"devices": {self.name: config}, "write_config": True})
+        msg = cmd.as_msg()
+        self._node_command_pub.publish(msg)
+        # TODO: properly wait for io command acknowleged using cmd timestamp
+        if timeout:
+            # add short delay for the reconfigure process to start
+            rospy.sleep(0.5)
+            intera_dataflow.wait_for(
+                lambda: self.is_ready() or self.needs_init(), timeout=timeout,
```

Comment:
Should this be `timeout - 0.5` to account for the sleep?

Reply:
oh, good call